src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -2768,38 +2768,58 @@ internal bool LoadModuleManifestData(ExternalScriptInfo scriptInfo, ManifestProc
             }
 
             if (scriptsToProcess != null)
-            {
-                foreach (string scriptFile in scriptsToProcess)
-                {
-                    bool found = false;
-                    PSModuleInfo module = LoadModule(scriptFile,
-                        moduleBase,
-                        string.Empty, // prefix (-Prefix shouldn't be applied to dot sourced scripts)
-                        null,
-                        ref options,
-                        manifestProcessingFlags,
-                        out found);
-
-                    // If we're in analysis, add the detected exports to this module's
-                    // exports
-                    if (found && (ss == null))
+            {
+                Version savedBaseMinimumVersion = BaseMinimumVersion;
+                Version savedBaseMaximumVersion = BaseMaximumVersion;
+                Version savedBaseRequiredVersion = BaseRequiredVersion;
+                Guid? savedBaseGuid = BaseGuid;
+
+                try
+                {
+                    BaseMinimumVersion = null;
+                    BaseMaximumVersion = null;
+                    BaseRequiredVersion = null;
+                    BaseGuid = null;
+
+                    foreach (string scriptFile in scriptsToProcess)
                     {
-                        foreach (string detectedCmdlet in module.ExportedCmdlets.Keys)
-                        {
-                            manifestInfo.AddDetectedCmdletExport(detectedCmdlet);
-                        }
+                        bool found = false;
+                        PSModuleInfo module = LoadModule(scriptFile,
```

Comment:
That is actually existing code; I didn't change it; looks like Github's diff'er has room for improvement :)  - see picture below.
Anyway, there are multiple LoadModule calls like this one throughout the file, so if refactoring is to be done - it has to be done to all of them - and that is out of scope of this bugfix.

![diff](https://cloud.githubusercontent.com/assets/11860095/26658552/94270334-461f-11e7-9439-d330bd64f5d3.png)